notebooks/smsnet_ensemble.ipynb

```diff
@@ -0,0 +1,493 @@
+    "    net, results = train_segmentation(\n",
+    "        net,\n",
+    "        train_loader,\n",
+    "        val_loader,\n",
+    "        model_parameters['num_epochs'],\n",
+    "        criterion,\n",
+    "        optimizer,\n",
+    "        device,\n",
+    "        savepath=parameters['save_path'],\n",
+    "        saveevery=None,\n",
+    "        scheduler=None,\n",
+    "        show=0,\n",
+    "        use_amp=False,\n",
+    "        clip_value=None\n",
+    "    )\n",
+    "    # Save network parameters\n",
+    "    model_params_path = f\"{parameters['save_path']}/{parameters['uid']}_SMSNet{idx}.pt\"\n",
+    "    net.save_network_parameters(model_params_path)\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Inference"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from dlsia.core.networks.baggins import model_baggin\n",
+    "from dlsia.core.networks.smsnet import SMSNetwork_from_file\n",
+    "from    qlty.qlty2D         import  NCYXQuilt"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "dataset = TiledDataset(\n",
+    "        data_tiled_uri=parameters['data_tiled_uri'],\n",
+    "        mask_idx=parameters['mask_idx'], # Keeping this for a quick inference for now, in future this will be out with updates from app.\n",
+    "        data_tiled_api_key=parameters['data_tiled_api_key'],\n",
+    "        qlty_window=model_parameters['qlty_window'],\n",
+    "        qlty_step=model_parameters['qlty_step'],\n",
+    "        qlty_border=model_parameters['qlty_border'],\n",
+    "        transform=transforms.ToTensor()\n",
+    "        )\n",
+    "\n",
+    "# Set Dataloader parameters (Note: we randomly shuffle the training set upon each pass)\n",
+    "inference_loader_params = {'batch_size': model_parameters['batch_size_inference'],\n",
+    "                            'shuffle': model_parameters['shuffle_inference']}\n",
+    "# Build Dataloaders\n",
+    "inference_loader = DataLoader(dataset, **inference_loader_params, collate_fn=custom_collate)\n",
+    "\n",
+    "device = torch.device(\"cuda:0\" if torch.cuda.is_available() else \"cpu\")\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "filenames = ['results/models/uid0013_SMSNet0.pt', 'results/models/uid0013_SMSNet1.pt', 'results/models/uid0013_SMSNet2.pt']\n",
+    "list_of_smsnet = []\n",
+    "for network in filenames:\n",
+    "    list_of_smsnet.append(SMSNetwork_from_file(network))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def segment(net, device, inference_loader, qlty_object):\n",
```

Comment:
Same here, this function appears to be defined elsewhere already and could be imported.

Reply:
Same as above.